recipes-bsp/bootfiles/rpi-config_git.bb

```diff
@@ -178,13 +178,26 @@ do_deploy() {
     fi
 
     # UART support
-    if [ "${ENABLE_UART}" = "1" ] || [ "${ENABLE_UART}" = "0" ] ; then
+    if [ "${ENABLE_UART}" = "1" ] || [ "${ENABLE_UART}" = "0" ]; then
         echo "# Enable UART" >>$CONFIG
         echo "enable_uart=${ENABLE_UART}" >>$CONFIG
     elif [ -n "${ENABLE_UART}" ]; then
         bbfatal "Invalid value for ENABLE_UART [${ENABLE_UART}]. The value for ENABLE_UART can be 0 or 1."
     fi
 
+    # U-Boot requires "enable_uart=1" for various boards to operate correctly
```

Comment:
This looks a bit redundant. Can we simplify it by just bbfatal (without any config changes) when we detect a misconfiguration? In this way we flag and users can fix in their conf.

```
if [ "${RPI_USE_U_BOOT}" = "1" ] && [ "${ENABLE_UART}" != "1" ]; then
  case "${UBOOT_MACHINE}" in
    rpi_0_w_defconfig|rpi_3_32b_config|rpi_4_32b_config|rpi_arm64_config)
      bbfatal "Invalid configuration: RPI_USE_U_BOOT requires to enable the UART in config.txt for ${MACHINE}"
      ;;
  esac
fi
```

Reply:
I see. This makes sense now on a second review. I'm okay with it now. Could you tweak git log a bit, as per the other comment?